File: src/main/java/me/udnek/fnafu/mechanic/camera/Camera.java
package me.udnek.fnafu.mechanic.camera;

import me.udnek.fnafu.map.Originable;
import me.udnek.fnafu.map.location.LocationSingle;
import org.bukkit.Location;

public class Camera implements Originable {

    private final LocationSingle locationSingle;
    private final String id;
    private final int tabletMenuPosition;

    public Camera(LocationSingle location, String id, int tabletMenuPosition){
        locationSingle = location;
        this.id = id;
        this.tabletMenuPosition = tabletMenuPosition;
    }

    public String getId() {return id;}
    public int getTabletMenuPosition() {return tabletMenuPosition;}
    public LocationSingle getLocation() {return locationSingle;}
    @Override
    public void setOrigin(Location location) {
        locationSingle.setOrigin(location);
    }
}

File: src/main/java/me/udnek/fnafu/mechanic/camera/Camera.kt
package me.udnek.fnafu.mechanic.camera

import me.udnek.fnafu.util.Originable
import me.udnek.fnafu.map.location.LocationSingle
import org.bukkit.*

class Camera : Originable {
    val location: LocationSingle
    val id: String
    val tabletMenuPosition: Int
    val rotationAngle: Float

    constructor(location: LocationSingle, id: String, tabletMenuPosition: Int, rotationAngle: Float) {
        this.location = location
        this.id = id
        this.tabletMenuPosition = tabletMenuPosition
        this.rotationAngle = rotationAngle
    }

    override fun setOrigin(origin: Location) {
        location.setOrigin(origin)
    }
}
